worker-batch-ingestion-plugin/src/main/java/com/github/cafdataprocessing/worker/ingestion/IngestionBatchWorkerPlugin.java

```diff
@@ -130,6 +130,13 @@ private void handleSingleBatchId(final String batchId, final BatchWorkerServices
             final TenantId tenantId = new TenantId(extractTenantId(batchId));
             final BatchId batchIdExtracted = extractBatchIds(batchId).get(0);
             final Path pathOfSubBatches = fileSystemProvider.getPathForBatch(tenantId, batchIdExtracted);
+
+            if (!Files.exists(pathOfSubBatches)) {
+                log.error("Exception while reading the batch folder: " + pathOfSubBatches + ", the directory does not exist");
+                throw new BatchDefinitionException("Exception while reading the batch folder: " + pathOfSubBatches
+                    + ", the directory does not exist");
```

Comment:
I don't like the fact that folders are mentioned here.  Let's just say that the specified batch was not found, or something along those lines.